llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -998,18 +998,18 @@ uint64_t InstrProfRecord::remapValue(uint64_t Value, uint32_t ValueKind,
 }
 
 void InstrProfRecord::addValueData(uint32_t ValueKind, uint32_t Site,
-                                   InstrProfValueData *VData, uint32_t N,
+                                   ArrayRef<InstrProfValueData> VData,
                                    InstrProfSymtab *ValueMap) {
-  for (uint32_t I = 0; I < N; I++) {
-    VData[I].Value = remapValue(VData[I].Value, ValueKind, ValueMap);
-  }
   std::vector<InstrProfValueSiteRecord> &ValueSites =
       getOrCreateValueSitesForKind(ValueKind);
   assert(ValueSites.size() == Site);
-  if (N == 0)
+  if (VData.empty())
     ValueSites.emplace_back();
-  else
-    ValueSites.emplace_back(VData, VData + N);
+  else {
+    ValueSites.emplace_back(VData.begin(), VData.end());
+    for (auto &V : ValueSites.back().ValueData)
```

Comment:
Before or after this PR, this part has dense information for readers to figure out what's going on.

As I understand it, `ValueSites.emplace_back` constructs one vector element using ArrayRef, and the for-loop iterates a data member of this vector element.

Would you mind adding some comments ?

Reply:
I hope the latest iteration simplifies things a bit.  We remap values first in a temporary vector.  Then we add the vector to the value site with the move semantics.